cupyx/jit/_compile.py

```diff
@@ -768,7 +772,8 @@ def _indexing(array, index, env):
     if isinstance(array.ctype, _cuda_types.Tuple):
         if is_constants(index):
             i = index.obj
-            return Data(f'thrust::get<{i}>({array.code})', array.types[i])
+            t = array.ctype.types[i]
+            return Data(f'thrust::get<{i}>({array.code})', t)
```

Comment:
This is a bug fix which I am not sure if we should backport, because apparently it's not caught by the test suite.